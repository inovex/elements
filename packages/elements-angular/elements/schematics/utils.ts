import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
