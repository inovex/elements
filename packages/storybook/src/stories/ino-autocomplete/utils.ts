import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
