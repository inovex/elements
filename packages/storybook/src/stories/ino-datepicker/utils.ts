import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
