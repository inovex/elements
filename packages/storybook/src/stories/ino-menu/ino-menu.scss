.story-ino-menu {
  padding: 100px;

  .row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }
}
