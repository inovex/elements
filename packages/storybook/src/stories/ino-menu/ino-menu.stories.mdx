import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-menu/readme.md";

<Meta title="Structure/ino-menu/Documentation" />

<Description markdown={readme} />
