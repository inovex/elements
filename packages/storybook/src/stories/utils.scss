/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
