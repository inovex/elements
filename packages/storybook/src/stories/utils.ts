import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
