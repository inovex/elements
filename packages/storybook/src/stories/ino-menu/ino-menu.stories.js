import { boolean } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-menu/readme.md';
import './ino-menu.scss';

const BTN_ID_CUSTOM_MENU = 'btn-custom-menu';
const BTN_ID_DIVIDED_MENU = 'btn-divided-menu';

const ID_CUSTOM_MENU = 'custom-menu';
const ID_DIVIDED_MENU = 'divided-menu';

function subscribeToComponentEvents() {
  // == event block
  const clickHandler = function (e) {
    const targetId = e.target.id;

    if (targetId !== BTN_ID_CUSTOM_MENU && targetId !== BTN_ID_DIVIDED_MENU) {
      return;
    }

    const menuEl = document.getElementById(
      targetId === BTN_ID_CUSTOM_MENU ?
        ID_CUSTOM_MENU : ID_DIVIDED_MENU
    );

    if (!menuEl) return;

    const isMenuOpen = menuEl.getAttribute('ino-open');
    menuEl.setAttribute(
      'ino-open',
      isMenuOpen === 'false' ? 'true' : 'false'
    );
  };

  function closeHandler(evt) {
    evt.target.setAttribute('ino-open', 'false');
  }

  document.addEventListener('click', clickHandler);
  document.addEventListener('menuClose', closeHandler)

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('click', clickHandler);
    document.removeEventListener('menuClose', closeHandler)
  };
}

export default {
  title: 'Structure/<ino-menu>',

  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-menu">
      <h4>Customizable Menu</h4>
      <ino-button id="${BTN_ID_CUSTOM_MENU}">Open menu</ino-button>
      <ino-menu
        id="${ID_CUSTOM_MENU}"
        ino-for="button"
        ino-open="${boolean('ino-open', false)}"
      >
        <ino-list-item tabindex="0" ino-text="Item"></ino-list-item>
        <ino-list-item tabindex="0" ino-text="Item 2"></ino-list-item>
      </ino-menu>

      <h4>Variation with divider</h4>
      <ino-button id="${BTN_ID_DIVIDED_MENU}">Open divided menu</ino-button>
      <ino-menu id="${ID_DIVIDED_MENU}" ino-for="menu-2" ino-open="${boolean(
        'ino-open',
        false
      )}">
        <ino-list-item ino-text="Home"></ino-list-item>
        <ino-list-item ino-text="Projects"></ino-list-item>
        <ino-list-divider></ino-list-divider>
        <ino-list-item ino-text="User"></ino-list-item>
        <ino-list-item ino-text="Settings"></ino-list-item>
      </ino-menu>
    </div>
  `;
