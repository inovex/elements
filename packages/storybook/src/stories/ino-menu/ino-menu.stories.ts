import { html } from 'lit-html';
import './ino-menu.scss';
import { defaultDecorator } from '../utils';

export default {
  title: `Structure/ino-menu`,
  component: 'ino-menu',
  decorators: [(story) => defaultDecorator(story, 'story-ino-menu')],
};

export const Playground = (args) => html`
  <ino-card ino-disable-elevation>
    <div class="row" slot="header">
      <p>Open Menu</p>
      <ino-icon-button icon="menu">
        <ino-menu placement="${args.placement}">
          <ino-list-item tabindex="0" text="Item 1"></ino-list-item>
          <ino-list-item tabindex="0" text="Item 2"></ino-list-item>
        </ino-menu>
      </ino-icon-button>
    </div>
  </ino-card>
`;
Playground.args = {
  placement: 'auto',
};
Playground.argTypes = {
  placement: {
    type: 'select',
  },
  options: ['auto', 'top', 'right', 'bottom', 'left'],
};
