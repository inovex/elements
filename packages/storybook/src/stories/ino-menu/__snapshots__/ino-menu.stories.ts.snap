// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-menu Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-menu">
    <ino-card disable-elevation
              class="hydrated"
    >
      <div class="mdc-card mdc-card--outlined">
        <div class="ino-card__checkmark-container">
        </div>
        <div class="ino-card__header">
          <div class="row"
               slot="header"
          >
            <p>
              Open Menu
            </p>
            <ino-icon-button icon="menu"
                             id="elements-menu_8vywhm2c0"
                             class="ino-icon-button--color-scheme-primary hydrated"
                             aria-expanded="false"
            >
              <ino-menu placement="auto"
                        class="hydrated"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div>
                  </div>
                </ino-popover>
              </ino-menu>
              <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                      type="button"
                      style="--mdc-ripple-fg-size:28px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:10px; --mdc-ripple-top:10px;"
              >
                <div class="mdc-icon-button__ripple">
                </div>
                <span class="mdc-icon-button__icon">
                  <ino-icon class="hydrated ino-icon--color-scheme-primary">
                  </ino-icon>
                </span>
              </button>
            </ino-icon-button>
          </div>
        </div>
      </div>
    </ino-card>
  </div>
</div>
`;
